File: programmers/lvl_1/12982.js
// 예산

function solution(d, budget) {
  var answer = 0;
  if (d.reduce((acc, cur) => acc + cur, 0) <= budget) {
    return d.length;
  } else {
    d.sort((a, b) => a - b);
    // console.log(d);
    for (amount of d) {
      //   console.log(budget);
      budget -= amount;
      if (budget < 0) {
        break;
      } else if (budget >= 0) {
        answer++;
      }
    }
    return answer;
  }
}

d = [1, 3, 2, 5, 4];
budget = 9;
// d = [2, 2, 3, 3];
// budget = 10;

a = solution(d, budget);
console.log(a);

File: programmers/lvl_1/12982.py
# 예산


def solution(d, budget):
    answer = 0
    if sum(d) <= budget:
        return len(d)
    else:
        d.sort()
        for amount in d:
            budget -= amount
            if budget < 0:
                break
            else:
                answer += 1

        return answer


d = [1, 3, 2, 5, 4]
budget = 9
# d = [2, 2, 3, 3];
# budget = 10;

a = solution(d, budget)
print(a)
